### PR TITLE
[cnd] Allow configuration of Bitcoin's fee parameter.

### PR DESCRIPTION
**Pull request by D4nte**

### CODE REVIEW

File: cnd/src/http_api/hbit_herc20/alice.rs [hunk 1]
```diff
@@ -40,7 +40,10 @@ impl RedeemAction
 {
     type Output = ethereum::CallContract;
 
-    fn redeem_action(&self) -> anyhow::Result<Self::Output> {
+    fn redeem_action(
+        &self,
+        _bitcoin_fee_rate_per_vbyte: bitcoin::Amount,
```
Comment (thomaseizinger) [inline cnd/src/http_api/hbit_herc20/alice.rs:45]: Leaving as `_` might bring it onto one line again :)

```suggestion
        _: bitcoin::Amount,
```

File: cnd/src/http_api/route_factory.rs [hunk 2]
```diff
@@ -45,6 +45,11 @@ pub fn create(
         }
     };
 
+    let bitcoin_fee_rate_per_vbyte = warp::any().map({
+        let bitcoin_fee_rate_per_vbyte = settings.bitcoin.fees.sat_per_vbyte;
+        move || bitcoin_fee_rate_per_vbyte
+    });
```
Comment (thomaseizinger) [inline cnd/src/http_api/route_factory.rs:51]: In other routes, I just passed the complete settings previously. That results in less code churn if we wanted to access other things as well in those routes.

File: cnd/src/http_api/swaps.rs [hunk 3]
```diff
@@ -69,6 +70,7 @@ async fn handle_get_swap(
     id: LocalSwapId,
     storage: Storage,
     connectors: Connectors,
+    bitcoin_fee_rate_per_vbyte: ::bitcoin::Amount,
```
Comment (thomaseizinger) [inline cnd/src/http_api/swaps.rs:73]: Maybe we could shorten this variable to something like this? The `per` already gives away that it is a rate.

```suggestion
    btc_per_vbyte: ::bitcoin::Amount,
```

File: cnd/src/main.rs [hunk 4]
```diff
@@ -135,7 +137,12 @@ fn main() -> anyhow::Result<()> {
         .build()?;
 
     let bitcoin_connector = {
-        let config::Bitcoin { bitcoind, network } = &settings.bitcoin;
+        let config::Bitcoin {
+            bitcoind,
+            network,
+            // TODO
```
Comment (thomaseizinger) [inline cnd/src/main.rs:143]: What is this TODO about?

File: cnd/src/serde.rs [hunk 5]
```diff
@@ -0,0 +1,26 @@
+use serde::{Deserialize, Deserializer, Serializer};
+
+pub mod bitcoin_amount {
```
Comment (thomaseizinger) [inline cnd/src/serde.rs:3]: Should this move into the `comit` lib, next to the other serde module?

**Review by thomaseizinger — APPROVED** [review]

**Comment by D4nte** [comment]
bors retry

**Pull request merged**